### PR TITLE
Enable Passphrase for Restore Command

### LINKED ISSUE #1671: Restore command does not support passphrase
## Current Behavior
No method available to pass a passphrase when restoring a wallet from Mnemonic

## Possible Solution
Addition of a command-line flag/option to allow (optionally) passing a passphrase.

## Steps to Reproduce
1. execute `ord wallet restore --help`
2. observe the lack of passphrase options/flags

## Possible Implementation
Example solution implementation is available in #1669

### PR DESCRIPTION
**Pull request by Psifour**
## Description
Implements an option to pass a passphrase when restoring from mnemonic.

## Related Issue
Closes #1671 

## Motivation and Context
Without this change users are only able to (easily) import a wallet from Sparrow if they did NOT use a passphrase.

## How Has This Been Tested?
Default case has been verified to call to_seed() with no passphrase and all single word data passed with the option is verified to replace that. Utilizes the standardized behavior as seen in other sub-commands.

### CODE REVIEW

File: src/subcommand/wallet/restore.rs [hunk 1]
```diff
@@ -4,11 +4,16 @@ use super::*;
 pub(crate) struct Restore {
   #[clap(help = "Restore wallet from <MNEMONIC>")]
   mnemonic: Mnemonic,
+  #[clap(long, short = 'p', help = "Use <PASSPHRASE> when deriving wallet")]
```
Comment (raphjaph) [inline src/subcommand/wallet/restore.rs:7]:
```suggestion
  #[clap(long, help = "Use <PASSPHRASE> when deriving wallet")]
```
We only use long flags at the moment so let's stay consistent with the others.

**Review by raphjaph — CHANGES_REQUESTED** [review]
Good idea!

Would be great to also add this functionality to `ord wallet create` and write integration tests that both work. Integration tests can be found in `tests/wallet/{create, restore}.rs`.

File: src/subcommand/wallet/restore.rs (hunk 1 above)
Comment (Psifour) [reply src/subcommand/wallet/restore.rs:7]: Will comply

**Comment by Psifour** [comment]
I've already done the create command changes, but have not implemented unit tests. Will try to get my work committed tonight or tomorrow on this.

**Review by raphjaph — APPROVED** [review]
This looks great!

I'll just wait for @casey to have a glance at this before merging.

File: src/subcommand/wallet/create.rs [hunk 2]
```diff
@@ -3,17 +3,31 @@ use super::*;
 #[derive(Serialize)]
 struct Output {
   mnemonic: Mnemonic,
+  passphrase: Option<String>,
+}
```
Comment (raphjaph) [inline src/subcommand/wallet/create.rs:7]:
```suggestion
}

```

File: src/subcommand/wallet/create.rs [hunk 3]
```diff
@@ -3,17 +3,32 @@ use super::*;
...
+  passphrase: Option<String>,
 }
 
-pub(crate) fn run(options: Options) -> Result {
-  let mut entropy = [0; 16];
-  rand::thread_rng().fill_bytes(&mut entropy);
+#[derive(Debug, Parser)]
+pub(crate) struct Create {
+  #[clap(long, help = "Use <PASSPHRASE> when deriving wallet")]
+  pub(crate) passphrase: Option<String>,
```
Comment (raphjaph) [inline src/subcommand/wallet/create.rs:12]:
```suggestion
  #[clap(long, default_value = "", help = "Use <PASSPHRASE> to derive wallet seed.")]
  pub(crate) passphrase: String,
```

File: src/subcommand/wallet/create.rs [hunk 4]
```diff
@@ -3,17 +3,32 @@ use super::*;
...
   mnemonic: Mnemonic,
+  passphrase: Option<String>,
 }
 
-pub(crate) fn run(options: Options) -> Result {
-  let mut entropy = [0; 16];
-  rand::thread_rng().fill_bytes(&mut entropy);
+#[derive(Debug, Parser)]
+pub(crate) struct Create {
+  #[clap(long, default_value = "", help = "Use <PASSPHRASE> to derive wallet seed.")]
```
Comment (raphjaph) [inline src/subcommand/wallet/create.rs:11]: We are using the specific language of BIP 39 here.

File: src/subcommand/wallet/create.rs [hunk 5]
```diff
@@ -3,17 +3,32 @@ use super::*;
...
+    let mut entropy = [0; 16];
+    rand::thread_rng().fill_bytes(&mut entropy);
 
-  let mnemonic = Mnemonic::from_entropy(&entropy)?;
+    let mnemonic = Mnemonic::from_entropy(&entropy)?;
 
-  initialize_wallet(&options, mnemonic.to_seed(""))?;
+    initialize_wallet(
+      &options,
+      mnemonic.to_seed(self.passphrase.clone().unwrap_or("".into())),
```
Comment (raphjaph) [inline src/subcommand/wallet/create.rs:24]:
```suggestion
      mnemonic.to_seed(self.passphrase),
```

File: tests/wallet/restore.rs [hunk 6]
```diff
@@ -20,3 +20,31 @@ fn restore_generates_same_descriptors() {
 
   assert_eq!(rpc_server.descriptors(), descriptors);
 }
+
+#[test]
+fn restore_generates_same_descriptors_with_passphrase() {
+  let passphrase = "unittest";
```
Comment (raphjaph) [inline tests/wallet/restore.rs:26]:
```suggestion
  let passphrase = "foo";
```

**Review by raphjaph — APPROVED** [review]
Small changes. Will merge now!

**Pull request merged**